### PR TITLE
enable typescript strict mode

### PR DESCRIPTION
**Pull request by promer94**

### CODE REVIEW

File: jest.config.js [hunk 1]
```diff
@@ -2,5 +2,10 @@ module.exports = {
   preset: 'ts-jest',
   testRegex: '/test/.*\\.test\\.tsx$',
   modulePathIgnorePatterns: ['<rootDir>/examples/'],
-  setupFilesAfterEnv: ['<rootDir>/jest-setup.ts']
+  setupFilesAfterEnv: ['<rootDir>/jest-setup.ts'],
+  globals: {
+    'ts-jest': {
+      diagnostics: false
```
Comment (huozhi) [inline jest.config.js:8]: `diagnostics` is removed in #903, I think we want to keep it removed

Comment (promer94) [reply jest.config.js]: find a batter way to disable strict mode in test 👌

**Comment by koba04** [comment]
Great work! I prefer `@ts-expect-error` to `@ts-ignore`, but the version SWR uses doesn't support it, so it would be great if it is replaced after updating the version.

File: src/index.ts [hunk 2]
```diff
@@ -11,7 +11,7 @@ export {
   revalidateType,
   RevalidateOptionInterface,
   keyInterface,
-  responseInterface,
+  ResponseInterface as responseInterface,
```
Comment (shuding) [inline src/index.ts:14]: This is a breaking change, let’s do it in #1016  instead.

**Review by shuding — COMMENTED** [review]
Generally looking great, nice work! 🎉

I might ship #1016 before this one, because the types are broken in the current beta release. But I think there won’t be too many conflicts.

File: src/index.ts (hunk 2 above)
Comment (shuding) [reply src/index.ts:14]: #1016 was merged so this PR is unblocked.

**Review by shuding — APPROVED** [review]
Thank you so much!  🎉 🎉

**Pull request merged**